### PR TITLE
Add DCI linting to CI of PRs

### PR DESCRIPTION
**Pull request by aklenik**
As recommended by the project update template and https://github.com/petermetz/gh-action-dci-lint
Let's see whether the remnant phrases in the changelog and third-party artifacts will trigger the alarm before making this a branch protection rule.

Signed-off-by: Attila Klenik <a.klenik@gmail.com>

### CODE REVIEW

File: packages/caliper-core/test/worker/tx-observers/prometheus-push-tx-observer.js [hunk 1]
```diff
@@ -216,7 +216,8 @@ describe('When using a PrometheusPushTxObserver', () => {
 
         await prometheusPushTxObserver.deactivate();
 
-        // Values should be zero, or empty (https://github.com/siimon/prom-client/blob/master/test/counterTest.js)
+        // Values should be zero, or empty 
+        // Ref: https://github.com/siimon/prom-client/  file test/counterTest.js
```
Comment (davidkel) [inline packages/caliper-core/test/worker/tx-observers/prometheus-push-tx-observer.js:220]: this url doesn't look valid. If we can't use the real url then we should delete the line

Comment (aklenik) [reply packages/caliper-core/test/worker/tx-observers/prometheus-push-tx-observer.js:220]: It's split into repo and file parts. I could structure it better if needed. But the entire file URL contains `master`...
Or we can remove it entirely

**Review by davidkel — CHANGES_REQUESTED** [review]
just a couple of minor comments

File: packages/caliper-core/test/worker/tx-observers/prometheus-push-tx-observer.js (hunk 1 above)
Comment (aklenik) [reply packages/caliper-core/test/worker/tx-observers/prometheus-push-tx-observer.js:220]: @davidkel, could be structured like this:
`See the "test/counterTest.js" file in https://github.com/siimon/prom-client/`

**Review by davidkel — APPROVED** [review]
LGTM

**Pull request merged**